compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1163,6 +1163,426 @@ TR::Register *OMR::X86::TreeEvaluator::cstoreEvaluator(TR::Node *node, TR::CodeG
+
+   //Residual
+   generateLabelInstruction(LABEL, node, residualLabel, cg);
+   generateRegRegInstruction(MOVRegReg(), node, counterRegister, arrayLenRegister, cg);   
+   generateRegImmInstruction(ANDRegImm4(), node, counterRegister, 0x7, cg);
+   generateInstruction(REPECMPSB, node, deps, cg);
+   generateLabelInstruction(JNE4, node, notEqualLabel, cg);
+
+   //Equal
+   generateRegRegInstruction(MOVRegReg(), node, resultRegister, arrayLenRegister, cg);   
+   generateLabelInstruction(JMP4, node, endResultLabel, cg);
+   cg->stopUsingRegister(arrayLenRegister);
+
+   //Inequality found, figure out which byte
+   generateLabelInstruction(LABEL, node, backtrackLabel, cg);
+   generateRegImmInstruction(SUBRegImm4(), node, src1AddrRegister, byteLen, cg);
+   generateRegImmInstruction(SUBRegImm4(), node, src2AddrRegister, byteLen, cg);
+   generateRegImmInstruction(MOVRegImm4(), node, counterRegister, byteLen, cg);
+   generateInstruction(REPECMPSB, node, deps, cg);
+   cg->stopUsingRegister(counterRegister);
+
+   //NotEqual
+   generateLabelInstruction(LABEL, node, notEqualLabel, cg);
+   generateRegImmInstruction(SUBRegImm4(), node, src1AddrRegister, 1, cg);
+   generateRegRegInstruction(SUBRegReg(), node, src1AddrRegister, resultRegister, cg);
+   generateRegRegInstruction(MOVRegReg(), node, resultRegister, src1AddrRegister, cg);   
+   generateLabelInstruction(JMP4, node, endResultLabel, cg);
+
+   //Result 
+   cg->stopUsingRegister(src1AddrRegister);
+   cg->stopUsingRegister(src2AddrRegister);
+   deps->stopAddingConditions();
+
+   generateLabelInstruction(LABEL, node, endResultLabel, deps, cg);
+   endResultLabel->setEndInternalControlFlow();
+
+   node->setRegister(resultRegister);
+
+   cg->decReferenceCount(src1AddrNode);
+   cg->decReferenceCount(src2AddrNode );
+   cg->decReferenceCount(lengthNode );
+
+   return resultRegister;
+   }
+
+static TR::Register *ConstLenArraycmpEvaluator(TR::Node *node, TR::CodeGenerator *cg, uintptrj_t arrLen)
+   {
+   TR::Register *resultRegister = cg->allocateRegister(TR_GPR);
+   TR::Node *src1AddrNode = node->getChild(0);
+   TR::Node *src2AddrNode = node->getChild(1);
+   TR::Node *lengthNode = node->getChild(2);
+   
+   int32_t byteLen = 4;
+   if (TR::Compiler->target.is64Bit())
+      byteLen = 8;
+
+   TR::Register *tempSrc1AddrRegister = cg->evaluate(src1AddrNode);;
+   TR::Register *tempSrc2AddrRegister = cg->evaluate(src2AddrNode);
+
+   TR::Register *src1AddrRegister = cg->allocateRegister();
+   TR::Register *src2AddrRegister = cg->allocateRegister();
+
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *notEqualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *lessThanLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endResultLabel = generateLabelSymbol(cg);
+
+   //Move into registers to for use
+   generateRegRegInstruction(MOVRegReg(), node, src1AddrRegister, tempSrc1AddrRegister, cg);   
```

Comment:
These moves are done regardless of the lifetime of the original register (i.e., if the ref count is one you don't need to copy it).  I think a `gprClobberEvaluate` instead of an `evaluate` on these nodes is what you want, as you'll get the copy done for you.